service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/RespondentController.java

```diff
@@ -57,29 +66,36 @@ public AboutToStartOrSubmitCallbackResponse handleMidEvent(@RequestBody Callback
     @PostMapping("/about-to-submit")
     public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody CallbackRequest callbackRequest) {
         CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        List<Element<Respondent>> confidentialRespondents =
+            confidentialDetailsService.addPartyMarkedConfidentialToList(Respondent.class, caseData.getAllRespondents());
+
+        if (isNotEmpty(confidentialRespondents)) {
+            caseDetails.getData().put("confidentialRespondents", confidentialRespondents);
+        } else {
+            caseDetails.getData().remove("confidentialRespondents");
+        }
 
-        return respondentService.addHiddenValues(caseDetails);
+        caseDetails.getData().put("respondents1", respondentService.modifyHiddenValues(caseData.getAllRespondents()));
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(caseDetails.getData())
+            .build();
     }
 
-    @SuppressWarnings("unchecked")
     private List<String> validate(CaseDetails caseDetails) {
         ImmutableList.Builder<String> errors = ImmutableList.builder();
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
 
-        List<Map<String, Object>> respondentObject =
-            (List<Map<String, Object>>) caseDetails.getData().get("respondents1");
-
-        List<Respondent> respondents = respondentObject.stream()
-            .map(respondent ->
-                mapper.mapObject((Map<String, Object>) respondent.get("value"), Respondent.class))
-            .collect(toList());
-
-        if (respondents.stream()
+        caseData.getAllRespondents().stream()
+            .map(Element::getValue)
             .map(Respondent::getParty)
             .map(Party::getDateOfBirth)
             .filter(Objects::nonNull)
-            .anyMatch(dob -> dob.isAfter(LocalDate.now()))) {
-            errors.add("Date of birth cannot be in the future");
-        }
+            .filter(dob -> dob.isAfter(LocalDate.now()))
+            .findAny()
```

Comment:
just curious, why do we need to change from `anyMatch` to `findAny` when we already had `errors` populated in the former scenario?

Reply:
this was changed elsewhere and I thought it best to standardise. See ChildController